sway-core/src/language/parsed/expression/mod.rs

```diff
@@ -321,7 +321,6 @@ pub enum ReassignmentTarget {
 pub struct StructExpressionField {
     pub name: Ident,
     pub value: Expression,
-    pub(crate) span: Span,
```

Comment:
Good question! That was the span of the `field_name: <initialization>` used only in the error message shown in struct instantiation that the struct field does not exist. That error is changed to point only at the field name, since having the non-existing field there is actually the issue, and not the `<initialization>` which on it's own does not need to be erroneous at all. This focus on the field as the source of the issue is even more important in the cases where the `<initialization>` spans several lines like in:
```Sway
x: Struct {
  a: 123,
  b: 345,
  ...
}
```
In that case the error span was containing a lot of noise.

I've checked Rust errors and they also mostly focus on the field only, but it's not fully consistent.
Also, this approach, to point to the actual issue I see as a general guidance, also when formulating the error message text.

Long story short, once that error span was changed, it turned out the `pub(crate) span: Span` wasn't used anywhere else. It felt odd to delete it, I must admit. I thought at first marking it with `#[allow(dead_code)]` but in the end opted for removal. If we ever need it, it will be totally trivial to get it back.